Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.cpp

```diff
@@ -48,41 +49,56 @@ namespace AzToolsFramework
 
     void DPERowWidget::Clear()
     {
-        destroyLayoutContents(m_columnLayout);
-        destroyLayoutContents(m_childRowLayout);
+        // propertyHandlers own their widgets, so don't destroy them here. Set them free!
+        for (auto propertyWidgetIter = m_widgetToPropertyHandler.begin(), endIter = m_widgetToPropertyHandler.end();
+             propertyWidgetIter != endIter; ++propertyWidgetIter)
+        {
+            m_columnLayout->removeWidget(propertyWidgetIter->first);
+        }
+        m_widgetToPropertyHandler.clear();
```

Comment:
Mostly as a note to myself:
We could technically just clear `m_widgetToPropertyHandler` here and automatically remove the widgets from the layout, but there's a relatively high likelihood of instead returning handlers to a shared pool in the future, so the explicit layout cleanup here is helpful.